features/events/daemon/src/main/java/org/opennms/netmgt/eventd/XmlEventProcessor.java

```
@@ -0,0 +1,182 @@
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.opennms.core.xml.JaxbUtils;
+import org.opennms.netmgt.events.api.EventForwarder;
+import org.opennms.netmgt.xml.event.Event;
+import org.opennms.netmgt.xml.event.EventReceipt;
+import org.opennms.netmgt.xml.event.Log;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.InputSource;
+
+/**
+ * Provides the logic and context of execution necessary to actually process a
+ * client's event XML document. When a new stream handler is created and
+ * assigned to an execution context it will unmarshal the remote document. The
+ * events from the remote document are then passed to the registered event
+ * handlers. All successfully processed events are acknowledged to the client by
+ * the generation of an XML event receipt.
+ * 
+ * @author <a href="mailto:weave@oculan.com">Brian Weaver </a>
+ * @author <a href="http;//www.opennms.org">OpenNMS </a>
+ * 
+ */
+public class XmlEventProcessor implements Processor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(XmlEventProcessor.class);
+
+    private EventForwarder m_eventForwarder;
+
+    public EventForwarder getEventForwarder() {
+        return m_eventForwarder;
+    }
+
+    public void setEventForwarder(EventForwarder eventForwarder) {
+        m_eventForwarder = eventForwarder;
+    }
+
+    @Override
+    public void process(Exchange exchange) {
+
+        final InputStream stream = exchange.getIn().getBody(InputStream.class);
+
+        // Unmarshal the XML document
+        Log eventLog = null;
+        try {
+            eventLog = JaxbUtils.unmarshal(Log.class, new InputSource(stream));
+            LOG.debug("Event record converted");
+        } catch (final Throwable e) {
+            LOG.error("Could not unmarshal the XML record", e);
+            // TODO: Throw an exception or something?
```

Shouldn't we re-throw the exception here? We can the use Camel to keep track of the count of failed exchanges.